src/r1cs/mod.rs

```diff
@@ -584,6 +711,25 @@ impl<G: Group> AbsorbInROTrait<G> for RelaxedR1CSInstance<G> {
   }
 }
 
+/// Return an instance and witness, given a shape, ck, and the needed vectors.
+pub fn instance_and_witness<G: Group>(
+  shape: &R1CSShape<G>,
+  ck: &CommitmentKey<G>,
+  input_assignment: Vec<G::Scalar>,
+  aux_assignment: Vec<G::Scalar>,
+) -> Result<(R1CSInstance<G>, R1CSWitness<G>), NovaError> {
+  let W = R1CSWitness::<G>::new(shape, aux_assignment)?;
+  let comm_W = W.commit(ck);
+  let instance = R1CSInstance::<G>::new(shape, comm_W, input_assignment)?;
+
+  Ok((instance, W))
+}
```

Comment:
The refactor of r1cs_instance_and_witness in https://github.com/lurk-lab/arecibo/pull/121 should supersede this.